core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -645,12 +646,14 @@ Builder populate() {
       defineMethod(EXP, BuiltInMethod.EXP.method, NullPolicy.STRICT);
       defineMethod(POWER, BuiltInMethod.POWER.method, NullPolicy.STRICT);
       defineMethod(ABS, BuiltInMethod.ABS.method, NullPolicy.STRICT);
-      defineMethod(LOG2, BuiltInMethod.LOG2.method, NullPolicy.STRICT);
 
       map.put(LN, new LogImplementor());
       map.put(LOG, new LogImplementor());
       map.put(LOG10, new LogImplementor());
 
+      map.put(LOG_MYSQLSPARK, new LogMysqlSparkImplementor());
```

Comment:
The SQL function name that you are registering is `LOG`, which is the same as what is added on line 651. The `equals` and `hashCode` methods used for the key is from the class `SqlOperator` and will only use the name+kind.

The result is that you overwrite the map entry from line 651. One way around this is set `LOG_MYSQLSPARK` to a new anonymous inner subclass of `SqlBasicFunction`. You would need to increase visibility of the `SqlBasicFunction` constructor.

```
LOG_MYSQLSPARK = new SqlBasicFunction(...) {};
```

Reply:
@normanj-bitquill Yes, you are right, so I use the name LOGMYSQLSPARK. The method you mentioned is something I didn't expect. Let me try it.
Thank you for your reply

Reply:
@normanj-bitquill I'm very sorry for the late reply, I've been very busy recently. You are correct, but I have a question, but will this have any impact, because the test can pass. 
Please forgive me for ignoring this question before.

Reply:
fix, 

```
  public static final SqlFunction LOG_MYSQL =
      SqlBasicFunction.create(SqlKind.LOG,
          ReturnTypes.DOUBLE_FORCE_NULLABLE,
          OperandTypes.NUMERIC_OPTIONAL_NUMERIC);
```
Changing the parameter list can fix this situation

Reply:
@normanj-bitquill I'm very sorry for the late reply. I've been very busy recently. You are correct, but I have a question, why the test can pass

Thanks for reminding me